File: Sasquatch/SasquatchWatchObjC Extension/ExtensionDelegate.m
#import "ExtensionDelegate.h"

@implementation ExtensionDelegate

- (void)applicationDidFinishLaunching {
}

- (void)applicationDidBecomeActive {
}

- (void)applicationWillResignActive {
}

- (void)handleBackgroundTasks:(NSSet<WKRefreshBackgroundTask *> *)backgroundTasks {
  for (WKRefreshBackgroundTask *task in backgroundTasks) {

    // Check the Class of each task to decide how to process it
    if ([task isKindOfClass:[WKApplicationRefreshBackgroundTask class]]) {

      // Be sure to complete the background task once you’re done.
      WKApplicationRefreshBackgroundTask *backgroundTask = (WKApplicationRefreshBackgroundTask *)task;
      [backgroundTask setTaskCompleted];
    } else if ([task isKindOfClass:[WKSnapshotRefreshBackgroundTask class]]) {

      // Snapshot tasks have a unique completion call, make sure to set your expiration date
      WKSnapshotRefreshBackgroundTask *snapshotTask = (WKSnapshotRefreshBackgroundTask *)task;
      [snapshotTask setTaskCompletedWithDefaultStateRestored:YES estimatedSnapshotExpiration:[NSDate distantFuture] userInfo:nil];
    } else if ([task isKindOfClass:[WKWatchConnectivityRefreshBackgroundTask class]]) {

      // Be sure to complete the background task once you’re done.
      WKWatchConnectivityRefreshBackgroundTask *backgroundTask = (WKWatchConnectivityRefreshBackgroundTask *)task;
      [backgroundTask setTaskCompleted];
    } else if ([task isKindOfClass:[WKURLSessionRefreshBackgroundTask class]]) {

      // Be sure to complete the background task once you’re done.
      WKURLSessionRefreshBackgroundTask *backgroundTask = (WKURLSessionRefreshBackgroundTask *)task;
      [backgroundTask setTaskCompleted];
    } else {

      // make sure to complete unhandled task types
      [task setTaskCompleted];
    }
  }
}

@end

File: Sasquatch/SasquatchWatchSwift Extension/ExtensionDelegate.swift
import WatchKit

class ExtensionDelegate : NSObject, WKExtensionDelegate {

  func applicationDidFinishLaunching() {
  }

  func applicationDidBecomeActive() {
  }

  func applicationWillResignActive() {
  }

  func handle(_ backgroundTasks : Set<WKRefreshBackgroundTask>) {
    for
      task in backgroundTasks {

        // Use a switch statement to check the task type
        switch
          task {
          case let backgroundTask as WKApplicationRefreshBackgroundTask:

            // Be sure to complete the background task once you’re done.
            backgroundTask
                .setTaskCompleted() case let snapshotTask as WKSnapshotRefreshBackgroundTask :

                // Snapshot tasks have a unique completion call, make sure to set your expiration date
                snapshotTask
                .setTaskCompleted(restoredDefaultState
                                  : true, estimatedSnapshotExpiration
                                  : Date.distantFuture, userInfo
                                  : nil) case let connectivityTask as WKWatchConnectivityRefreshBackgroundTask :

                // Be sure to complete the connectivity task once you’re done.
                connectivityTask
                .setTaskCompleted() case let urlSessionTask as WKURLSessionRefreshBackgroundTask :

                // Be sure to complete the URL session task once you’re done.
                urlSessionTask
                .setTaskCompleted() default :

                // make sure to complete unhandled task types
                task.setTaskCompleted()
          }
      }
  }
}
